radiorec.py

```diff
@@ -123,15 +125,30 @@ def record(args):
     if streamurl.endswith('.m3u'):
         verboseprint('Seems to be an M3U playlist. Trying to parse...')
         pool = urllib3.PoolManager()
-        with pool.request('GET',streamurl) as remotefile:
-            for line in remotefile:
-                if not line.decode('utf-8').startswith('#') and len(line) > 1:
-                    tmpstr = line.decode('utf-8')
+        try:
+            remotefile = pool.request('GET', streamurl)
+        except MaxRetryError:
+            print('The URL of the station is not found! Check' + args.station + ' in the Settings!')
+            sys.exit()
+        if remotefile.status != 200:
+            print('The URL of the station is somehow faulty! Check' + args.station + ' in the Settings!')
+            sys.exit()
+        else:
+            for line in remotefile.data.decode().split():
+                if not line.startswith('#') and len(line) > 1 and line.endswith('mp3'):
+                    tmpstr = line
                     break
-        streamurl = tmpstr
+        if not len(tmpstr) > 1:
+            print('Could not find a mp3 stream')
+            sys.exit()
+        else:
+            streamurl = tmpstr
 
     verboseprint(print_time() + " ... Stream URL: " + streamurl)
     target_dir = os.path.expandvars(settings['GLOBAL']['target_dir'])
+    if not os.path.isdir(target_dir + os.sep):
```

Comment:
why `+ os.sep`? 
Furthermore we could (at least) try to create the directory with `os.makedirs` prior this check with `exist_ok=True`